test/resource_owner_password_grant_test.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@TestOn("vm")
+library resource_owner_password_grant_test;
+
+import 'dart:convert';
+import 'dart:io';
+import 'package:crypto/crypto.dart';
+
+import 'package:test/test.dart';
+import 'package:oauth2/oauth2.dart' as oauth2;
+
+final List<int> ERROR = UTF8.encode(JSON.encode({
+  "error": "unauthorized",
+  "error_description": "unauthorized",
+  "error_uri": "http://error.com"
+}));
+
+final List<int> SUCCESS = UTF8.encode(JSON.encode({
+  "access_token": "2YotnFZFEjr1zCsicMWpAA",
+  "token_type": "bearer",
+  "expires_in": 3600,
+  "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA",
+}));
+
+void main() {
+  group('basic', () {
+    HttpServer server;
+    var authorizationEndpoint =
+        Uri.parse('http://${InternetAddress.LOOPBACK_IP_V4.host}:8080');
+
+    setUp(() async {
+      server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 8080);
```

Comment:
FYI: hard-coding a port number in a test is a recipe for flakiness. These tests will fail if it's run while any other process has bound port 8080. Instead, pass port 0 which will tell the OS to automatically allocate a free port, and then access that port via `HttpServer.port`.